lib/cordova-plugins/factory.js

```diff
@@ -0,0 +1,182 @@
+const logger = require('../Logger');
+
+/**
+ * plugin mock factory
+ * to create a proper abstraction for the test, the test does not neet to know whrere it's mocks come from !
+ *
+ * * How to mock a plugin
+ * ----------------------
+ * 1. Write its name, need to match the cordova plugin name and the actual mock filename, into the `wdi5: plugins: <pluginname>` config.
+ * 1.1 This triggers the load of the javascript file
+ *
+ * 2. Implement the plugin mock to override the needed plugin javascript functions, and call the `registerPlugin` function of the plugin factory. The third parameter is the critical:
+ * The function passed to the register function will be executed in the browser's webapp context!
+ * 2.1 This will trigger an attach of the mock funtions to the web app browser context.
+ *
+ * 3. The mock function will be automatically called by the app
+ */
+module.exports = {
+    // Object of plugin names read from config
+    _pluginListConfig: {},
+    // instanziated list of plugins
+    _pluginList: {
+        android: {},
+        ios: {},
+        browser: {},
+        electron: {}
+    },
+    // index
+    _: null,
+    _context: null,
+    // name of platform under test
+    _platform: '',
+
+    /**
+     * call to trigger the init process
```

Comment:
missing param description for _, missing param for context

Reply:
``` 
/**
     * call to trigger the init process
     * @param {index} _ WDI5 framework index.js
     * @param {WebdriverIO.BrowserObject} context
     */